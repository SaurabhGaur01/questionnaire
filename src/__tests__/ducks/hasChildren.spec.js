import {
    setChildrenPresent,
    setChildrenNotPresent,
    hasChildrenReducer,
} from '../../ducks/userInput/hasChildren';

describe('hasChildren Duck Tests', () => {
    const mockObject = "mockValue";

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setChildrenNotPresent should match the snapshot' , () => {
            expect(setChildrenNotPresent()).toMatchSnapshot();
        });
        it('setChildrenPresent should match the snapshot' , () => {
            expect(setChildrenPresent(mockObject)).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(hasChildrenReducer()).toEqual("");
        });
        it('should return the passed state if called with no action' , () => {
            expect(hasChildrenReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(hasChildrenReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value which is passed' , () => {
            expect(hasChildrenReducer('', setChildrenPresent(mockObject))).toEqual(mockObject);
        });
        it('should clear the store when called with setChildrenNotPresent' , () => {
            expect(hasChildrenReducer('', setChildrenNotPresent())).toEqual("");
        });
    });
});