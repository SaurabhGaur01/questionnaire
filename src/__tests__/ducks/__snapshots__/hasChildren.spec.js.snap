// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hasChildren Duck Tests Action Generator setChildrenNotPresent should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/DONT_HAVE_CHILDREN",
}
`;

exports[`hasChildren Duck Tests Action Generator setChildrenPresent should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/HAVE_CHILDREN",
  "value": "mockValue",
}
`;
